src/pages/blog/streamlit-deployment-https-social-auth.md

```diff
@@ -0,0 +1,122 @@
+
+## Deployment of Streamlit Apps with Gimlet
+
+Deploying a Streamlit application can be done in X number of steps.
+
+You can get started with Gimlet by connecting your GitHub account. After that, add the repository that has your Streamlit application by clicking the **Import** button next to it. Save the added repo by clicking the **I am done importing** button. If you can’t find it on the top of the list, use the search bar.
+
+Navigate to the deployment setup screen by clicking on the card of your Streamlit application. Set up the deployment in the following way:
+
+- **Deployment template:** Web Application. This’ll allow you to select the Dockerfile container image setting.
+- **Container Image:** Using a Dockerfile. This method requires a Dockerfile. If you don’t have one, the example below can be tailored to your app or used without changes. If you’re not familiar with Dockerfiles, read the comments.
+
+```
+# app/Dockerfile
+
+# This Dockerfile uses python:3.9-slim, which is a lightweight implementation of the Python image. This’ll reduce image size.
+FROM python:3.9-slim
+
+# Sets the working directory inside the container to `/app`. It’s unlikely that you’ll need to adjust this.
+WORKDIR /app
+
+# Installs several packages and cleans up the apt cache. In some cases some of these packages might not be needed to run a Streamlit container.
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    curl \
+    software-properties-common \
+    git \
+    && rm -rf /var/lib/apt/lists/*
+
+# Copies everything from the root directory to the working directory of the container. This is usually used for development environments.
+COPY . .
+
+# Installs Python packages listed in the requirements.txt.
+RUN pip3 install -r requirements.txt
+
+# Exposes port 8501 for external access. This is typical for Streamlit applications.
+EXPOSE 8501
+
+# Checks the health of the Streamlit server.
+HEALTHCHECK CMD curl --fail http://localhost:8501/_stcore/health
+
+# Defines how to start the application. In this case the Streamlit application’s filename is st.py, so feel free to adjust it to your app’s name.
+ENTRYPOINT ["streamlit", "run", "st.py", "--server.port=8501", "--server.address=0.0.0.0"]
+```
+
+- **Registry:** Gimlet Registry. This is made by Gimlet for you by default.
+- **Port:** `8501`. This is the default exposed port for Streamlit apps.
+
+When all the setting changes are made like listed above, you can click the **Deploy** button. After that, logs will appear and your app will be set up soon. When deployment status turns `Running`, you can check out the app in your browser by clicking on the link next to the status. Click **Write configuration to Git** to save the deployed application and to be able to edit its settings later.
+
+## Fancy Things for your Streamlit App
+
+You can configure additional settings for your deployed Streamlit app, such as HTTPS certificates and social authentication.
+
+### HTTPS Certificate
```

Comment:
cert-manager is enabled by default on all gimlet environments.

We can consider it enabled for all our tutorials